File: src/main/java/cn/don9cn/blog/action/bussiness/articlecomment/ArticleCommentAction.java
package cn.don9cn.blog.action.bussiness.articlecomment;

import cn.don9cn.blog.model.bussiness.acticlecomment.ArticleComment;
import cn.don9cn.blog.model.bussiness.articleclassify.ArticleClassify;
import cn.don9cn.blog.service.bussiness.articlecomment.interf.ArticleCommentService;
import org.apache.log4j.Logger;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;


/**
 * @Author: liuxindong
 * @Description: 文章评论action
 * @Create: 2017/11/15 16:43
 * @Modify:
 */
@RestController
@RequestMapping(value = "/bussiness/articleComment")
public class ArticleCommentAction extends BaseAction<ArticleComment> {
	
    private static Logger logger = Logger.getLogger(ArticleCommentAction.class);

	@Autowired
	private ArticleCommentService articleCommentService;


	@Override
	@PostMapping({"","/insert/public"})
	protected Object baseInsert(ArticleComment articleComment) {
		return articleCommentService.baseInsert(articleComment);
	}

	@Override
	protected Object baseInsertBatch(List<ArticleComment> list) {
		return null;
	}

	@Override
	protected Object baseUpdate(ArticleComment articleComment) {
		return null;
	}

	@Override
	@DeleteMapping
	protected Object baseRemove(String code) {
		return articleCommentService.baseDeleteById(code);
	}

	@Override
	@DeleteMapping("/batch")
	protected Object baseRemoveBatch(String codes) {
		return articleCommentService.baseDeleteBatch(codes);
	}

	@Override
	protected Object baseFindById(String code) {
		return null;
	}

	@Override
	protected Object baseFindAll() {
		return null;
	}

	@Override
	protected Object baseFindListByParams(ArticleComment articleComment) {
		return null;
	}

	@Override
	@GetMapping("/page")
	protected Object baseFindByPage(int page,int limit,String startTime,String endTime,String orderBy,ArticleComment articleComment) {
		return articleCommentService.baseFindByPage(new PageResult<>(page,limit,startTime,endTime,orderBy,articleComment));
	}

	/**
	 * 获取留言树
	 * @return
	 */
	@GetMapping("/tree/public")
	public Object getTree(String articleCode) {
		return articleCommentService.getTree(articleCode);
	}
}

File: src/main/java/cn/don9cn/blog/action/bussiness/article/ArticleCommentAction.kt
package cn.don9cn.blog.action.bussiness.article

import cn.don9cn.blog.support.mongo.core.PageResult
import cn.don9cn.blog.action.BaseAction
import cn.don9cn.blog.model.bussiness.acticlecomment.ArticleComment
import cn.don9cn.blog.service.bussiness.ArticleCommentService
import cn.don9cn.blog.support.action.ActionMsg
import cn.don9cn.blog.support.action.ActionMsgHandler
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping(value = ["/bussiness/articleComment"])
open class ArticleCommentAction : BaseAction<ArticleComment>() {

    @Autowired
    private var articleCommentService: ArticleCommentService? = null


    @PostMapping("", "/insert/public")
    override fun baseInsert(t: ArticleComment): ActionMsg {
        return ActionMsgHandler.insert(articleCommentService!!.baseInsert(t))
    }

    override fun baseInsertBatch(list: List<ArticleComment>): ActionMsg {
        return ActionMsgHandler.insert(articleCommentService!!.baseInsertBatch(list))
    }

    override fun baseUpdate(t: ArticleComment): ActionMsg {
        return ActionMsgHandler.update(articleCommentService!!.baseUpdate(t))
    }

    @DeleteMapping
    override fun baseRemove(code: String): ActionMsg {
        return ActionMsgHandler.delete(articleCommentService!!.baseDeleteById(code))
    }

    @DeleteMapping("/batch")
    override fun baseRemoveBatch(codes: String): ActionMsg {
        return ActionMsgHandler.delete(articleCommentService!!.baseDeleteBatch(codes))
    }

    override fun baseFindById(code: String): ActionMsg {
        return ActionMsgHandler.find(articleCommentService!!.baseFindById(code))
    }

    override fun baseFindAll(): ActionMsg {
        return ActionMsgHandler.find(articleCommentService!!.baseFindAll())
    }

    override fun baseFindListByParams(t: ArticleComment): ActionMsg {
        return ActionMsgHandler.find(articleCommentService!!.baseFindListByParams(t))
    }

    @GetMapping("/page")
    override fun baseFindByPage(page: Int, limit: Int, startTime: String?, endTime: String?, orderBy: String?, t: ArticleComment): ActionMsg {
        return ActionMsgHandler.find(articleCommentService!!.baseFindByPage(PageResult(page, limit, startTime, endTime, orderBy, t)))
    }

    /**
     * 获取留言树
     * @return
     */
    @GetMapping("/tree/public")
    open fun getTree(articleCode: String): ActionMsg {
        return ActionMsgHandler.find(articleCommentService!!.getTree(articleCode))
    }
}